src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
This isn't atomic, i.e. the destination may not exist at the time of this check but could be added after and before the actual call.  Does that matter?